src/Tizen.NUI/src/public/BaseComponents/VectorGraphcis/Drawable.cs

```diff
@@ -0,0 +1,105 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+using System.ComponentModel;
+
+namespace Tizen.NUI.BaseComponents.VectorGraphics
+{
+    /// <summary>
+    /// Drawable is a object class for drawing a vector primitive.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Drawable : BaseHandle
+    {
+        /// <summary>
+        /// Creates an initialized drawable.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        private Drawable() {}
+
+        internal Drawable(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
+        /// <summary>
+        /// The transparency level [0 ~ 1.0], 0 means totally transparent, while 1 means opaque.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public float Opacity
+        {   
+            get {
+                return Interop.Drawable.GetOpacity(BaseHandle.getCPtr(this));
+            }
+            set {
+                Interop.Drawable.SetOpacity(BaseHandle.getCPtr(this), value);
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            }   
+        }
+
+        /// <summary>
+        /// Set the angle of rotation transformation.
+        /// </summary>
+        /// <param name="degree">The degree value of angle.</param>
+        /// <returns>True when it's successful. False otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Rotate(float degree)
+        {   
+            bool ret = Interop.Drawable.Rotate(BaseHandle.getCPtr(this), degree);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        /// <summary>
+        /// Set the scale value of scale transformation.
+        /// </summary>
+        /// <param name="factor">The scale factor value.</param>
+        /// <returns>True when it's successful. False otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Scale(float factor)
+        {   
+            bool ret = Interop.Drawable.Scale(BaseHandle.getCPtr(this), factor);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        /// <summary>
+        /// Set the matrix value for affine transform.
+        /// </summary>
+        /// <param name="matrix">The float type array of 3x3 matrix.</param>
+        /// <returns>True when it's successful. False otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Transform(float[] matrix)
```

Comment:
I am worrying that if user provide an array that has less than 9 items, the program will have unexpected memory atemption.
To prevent this, please check given matrix size and throw an `ArgumentException` when it is not met 9.

FYI, C# array has a `Length` property. You can check the size of given array with it.

Reply:
I added transform's exception handling. thank you